content/Hello yaaf - Juan Manuel Ramallo.md

```diff
@@ -0,0 +1,73 @@
+## Yet another active form
+
+Introducing [yaaf](https://github.com/rootstrap/yaaf), a gem to ease the usage of the form object pattern in rails apps.
+
+### ~The~ My pain points
+
+The form object pattern is widely used across Rails apps,
+and yet we tend to write these form objects in a different manner from project to project.
+
+Most of the time, they don't even have the same interface across form objects in the same app.
+
+We also tend to write validations twice, in the model and in the form object,
+just so that the form object has the errors and the view can display them.
+
+Well, I can say that most of our in-house form objects make well use of database transactions.
+But I bet we rarely provide the controller the ability to use a banged method,
+expecting it to raise an error if things go south.
+
+Business logic is present everywhere in your app, controllers, models, helpers (some hardcore scenarios might
+include rails initializers), but is that right? Many times we find ourselves with the need to add business logic
+around the creation of an object, such as sending an email, updating other records or even calling an external service.
+We know that the controller is not a good place, and the model would be terrible as well because it violates SRP.
+So we end up creating several service objects and throwing the business logic there as if it were a trashcan.
+
+### Instead of all of that, we can...
+
+use yaaf. yaaf solves all of that in an easy way, let me explain:
+
+- Same interface? yaaf will encourage you to use `save` and just `save` to, well you know,
+SAVE the data from your form into models.
+
+- Contextual validations only? Please do that, and keep data integrity validations where they belong,
```

Comment:
Not sure if it's clear that contextual validations should be in the form objects.. alternative:

```suggestion
- Contextual validations only? Please put them in your form objects, and keep data integrity validations where they belong,
```